cmd/vclusterctl/cmd/platform/share/vcluster.go

```diff
@@ -0,0 +1,121 @@
+package share
+
+import (
+	"context"
+	"fmt"
+
+	agentstoragev1 "github.com/loft-sh/agentapi/v4/pkg/apis/loft/storage/v1"
+	"github.com/loft-sh/api/v4/pkg/product"
+	loftctlUtil "github.com/loft-sh/loftctl/v4/pkg/util"
+	"github.com/loft-sh/log"
+	"github.com/loft-sh/vcluster/pkg/cli/flags"
+	"github.com/loft-sh/vcluster/pkg/platform"
+	"github.com/loft-sh/vcluster/pkg/projectutil"
+	"github.com/mgutz/ansi"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VClusterCmd holds the login cmd flags
+type VClusterCmd struct {
+	*flags.GlobalFlags
+
+	Project     string
+	Cluster     string
+	ClusterRole string
+	User        string
+	Team        string
+
+	Log log.Logger
+}
+
+// NewVClustersCmd creates a new command
+func NewVClusterCmd(globalFlags *flags.GlobalFlags) *cobra.Command {
+	cmd := &VClusterCmd{
+		GlobalFlags: globalFlags,
+		Log:         log.GetInstance(),
+	}
+
+	cobraCmd := &cobra.Command{
+		Use:   "vcluster" + loftctlUtil.VClusterNameOnlyUseLine,
+		Short: "Shares a vcluster with another Platform user or team",
+		Long: `
+##########################################################################
+#################### vcluster platform list vclusters ####################
+##########################################################################
+Shares a vcluster with another Platform user or team
+
+Example:
+vcluster platform share vcluster myvcluster
+vcluster platform share vcluster myvcluster --project myproject
+vcluster platform share vcluster myvcluster --project myproject --user admin
+##########################################################################
+	`,
+		Args: loftctlUtil.VClusterNameOnlyValidator,
+		RunE: func(cobraCmd *cobra.Command, args []string) error {
+			return cmd.Run(cobraCmd.Context(), args)
+		},
+	}
+
+	cobraCmd.Flags().StringVar(&cmd.Cluster, "cluster", "", "The cluster to use")
+	cobraCmd.Flags().StringVar(&cmd.Project, "project", "", "The project to use")
+	cobraCmd.Flags().StringVar(&cmd.ClusterRole, "cluster-role", "loft-cluster-space-admin", "The cluster role which is assigned to the user or team for that space")
+	cobraCmd.Flags().StringVar(&cmd.User, "user", "", "The user to share the space with. The user needs to have access to the cluster")
+	cobraCmd.Flags().StringVar(&cmd.Team, "team", "", "The team to share the space with. The team needs to have access to the cluster")
+
+	return cobraCmd
+}
+
+// TODO(johannesfrey): Move flags and functionality to pkg/cli
```

Comment:
Have they been moved out already?

Reply:
Not yet. Would like to do this in a follow-up, as we already have this inconsistency currently on main. While most of the non-platform related vcluster commands directly call out to `pkg/cli` (e.g. https://github.com/loft-sh/vcluster/blob/main/cmd/vclusterctl/cmd/connect.go#L96-L99), the already merged platform commands directly implement the run functionality in `cmd` (e.g. https://github.com/loft-sh/vcluster/blob/main/cmd/vclusterctl/cmd/platform/add/cluster.go).